cita-secp256k1/src/signature.rs

```diff
@@ -316,7 +317,8 @@ impl Sign for Signature {
     fn sign(privkey: &Self::PrivKey, message: &Self::Message) -> Result<Self, Self::Error> {
         let context = &SECP256K1;
         // no way to create from raw byte array.
-        let sec: &SecretKey = unsafe { mem::transmute(privkey) };
+        let sec: &SecretKey =
```

Comment:
What's the clippy reason?

Reply:
```
warning: transmute from a reference to a reference
   --> cita-secp256k1/src/signature.rs:319:40
    |
319 |         let sec: &SecretKey = unsafe { mem::transmute(privkey) };
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^ help: try: `&*(privkey as *const types::H256 as *const secp256k1::SecretKey)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#transmute_ptr_to_ptr
```
## What it does
Checks for transmutes from a pointer to a pointer, or from a reference to a reference.

## Why is this bad
Transmutes are dangerous, and these can instead be written as casts.

## Example
```
let ptr = &1u32 as *const u32;
unsafe {
    // pointer-to-pointer transmute
    let _: *const f32 = std::mem::transmute(ptr);
    // ref-ref transmute
    let _: &f32 = std::mem::transmute(&1u32);
}
// These can be respectively written:
let _ = ptr as *const f32
let _ = unsafe{ &*(&1u32 as *const u32 as *const f32) };
```